src/components/common/TLSConnectionForm/types.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InputFieldState } from '@devtron-labs/devtron-fe-common-lib'
+
+export interface TLSConfigDTO {
+    caData: string
+    tlsCertData: string
+    tlsKeyData: string
+}
+
+export interface TLSConnectionDTO {
+    enableTLSVerification: boolean
+    tlsConfig: TLSConfigDTO
+    /**
+     * Used in case when we have already saved data and we have'nt filled the fields
+     */
+    isCADataPresent: boolean
+    isTLSCertDataPresent: boolean
+    isTLSKeyDataPresent: boolean
```

Comment:
suggestion: these can be typed better

Reply:
Have not hard coded the values for these anywhere, and it is being extended for input state so may not be a good idea.